modules/survey_module/jsx/DirectEntry.js

```diff
@@ -0,0 +1,384 @@
+				name.push(this.getElementName(element.Elements[i]));
+			}
+		} else {
+			name = element.Name
+		}
+
+		return name;
+	}
+
+	nextPage() {
+
+		let page = 0;
+		let finalPage = false;
+		if(this.state.page != 0) {
+			page = this.state.InstrumentJSON.Elements[this.state.page].Subtest;
+		}
+
+		let data = {
+			"data" : this.state.pageValues,
+			"page" : page
+		};
+		const that = this;
+
+		if(this.state.page === this.state.InstrumentJSON.Elements.length - 1) {
+			data['FinalPage'] = true;
+			finalPage = true;
+		}
+
+		$.ajax({
+		    url : this.state.api_url,
+		    data : JSON.stringify(data),
+		    type : 'PUT',
+		    contentType : 'application/json',
+		    success : function(result){
+		        const page = that.state.page + 1;
+		        let InstrumentJSON;
+		        let reviewPage;
+
+		        if(finalPage) {
+		        	InstrumentJSON = that.state.InstrumentJSON;
+		        	reviewPage = JSON.parse(result);
+		        } else {
+		        	InstrumentJSON = JSON.parse(result);
+		        }
+
+		        that.setState({
+					page: page,
+					errors: {},
+					InstrumentJSON: InstrumentJSON,
+					ReviewData: reviewPage
+				});
+
+				that.setupPageValues(page);
+				window.scrollTo(0,0);
+		   },
+		}).fail((responseData) => {
+			if(responseData.status === 400) {
+				const response = JSON.parse(responseData.responseText)
+				this.setState({
+					errors: response
+				});
+				swal({
+					title: "Error",
+					text: "Please resolve page errors before continuing"
+				}, function(e){
+					$('html, body').animate({
+	                    scrollTop: $($(".questionError")[0]).offset().top - 100
+	                }, 100);
+				});
+			}
+		});
+
+		
+	}
+
+	prevPage() {
+		const page = this.state.page - 1;
+
+        this.setState({
+			page: page,
+			errors: {}
+		});
+		this.setupPageValues(page);
+
+		window.scrollTo(0,0);
+	}
+
+	updateAnswer(fieldName, value) {
+		let data = {};
+		data[fieldName] = value;
+
+		$.ajax({
+		    url : this.state.api_url,
+		    data : JSON.stringify(data),
+		    type : 'PATCH',
+		    contentType : 'application/json'
+		}); 
+
+		this.setState(function(state) {
+			let values = state.values;
+			let pageValues = state.pageValues;
+			let stats = state.completionStats;
+
+			if(values[fieldName] == null || values[fieldName] == '') {
+				stats.completed = stats.completed + 1;
+			} else if (value == null || value == '') {
+				stats.completed = stats.completed - 1;
+			}
+
+			values[fieldName] = value;
+			pageValues[fieldName] = value;
+			return {
+				values: values,
+				pageValues: pageValues,
+				completionStats: stats
+			}
+		});
+	}
+
+	submit() {
+		$.ajax({
+		    url : this.state.api_url,
+		    type : 'POST',
+		    contentType : 'application/json'
+		}); 
+	}
+
+	render() {
+		if(!this.state.InstrumentJSON.Elements){
+			// Since the Instrument data is set when the component is
+			// mounted we want to display nothing until it has been set
+			return (
+				<div></div>
```

Comment:
A loading span would be nice.

Reply:
I am making a `Loader` Component I hope to make a pull request for soon.